src/components/Footer.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useMemo } from 'react';
+import classNames from 'classnames';
+
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  visibleTodos: Todo[] | null
```

Comment:
I think you will never have `null` visibleTodos. You set an empty array by default, so here you will have an array always.
Check this case, please.